buildSrc/subprojects/uber-plugins/src/main/kotlin/gradlebuild/ci-lifecycle.gradle.kts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+tasks.registerEarlyFeedbackLifecycleTasks()
+tasks.named("quickTest") {
+    dependsOn("test")
+}
+tasks.named("platformTest") {
+    dependsOn("test")
+}
+tasks.configureCIIntegrationTestDistributionLifecycleTasks()
+tasks.configureCICrossVersionTestDistributionLifecycleTasks()
+
+val ciGroup = "CI Lifecycle"
+
+/**
+ * Basic compile and check lifecycle tasks.
+ */
+fun TaskContainer.registerEarlyFeedbackLifecycleTasks() {
+    register("compileAllBuild") {
+        description = "Initialize CI Pipeline by priming the cache before fanning out"
+        group = ciGroup
+        dependsOn("compileAll")
+    }
+
+    register("sanityCheck") {
+        description = "Run all basic checks (without tests) - to be run locally and on CI for early feedback"
+        group = "verification"
+        dependsOn("compileAll", "codeQuality")
+    }
+}
+
+fun TaskContainer.configureCIIntegrationTestDistributionLifecycleTasks() {
+    named("quickTest") {
+        dependsOn("embeddedIntegTest")
+    }
```

Comment:
I am not confident with the change because now `soak` tests can be triggered using `quickTest` lifecycle task, which is probably not semantically correct.

Reply:
I think it's ok.